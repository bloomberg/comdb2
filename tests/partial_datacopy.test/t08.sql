create table t {
    schema {
        int a
        vutf8 b[6]
        int c
        int d
    }
    keys {
        datacopy(b, c) "a" = a
    }
};$$


select * from comdb2_keys where tablename='t';
select * from comdb2_partial_datacopies where tablename='t';

insert into t with recursive r(n) as (select 1 union all select n + 4 from r where n < 97) select n, "cdb" || (n + 1), n + 2, n + 3 from r;
select * from t where a < 40;
select a, b, c from t where a < 40;

alter table t {
    schema {
        int a
        vutf8 b[4]
        int c
        int d
    }
    keys {
        datacopy(b, c) "a" = a
    }
};$$

select * from comdb2_keys where tablename='t';
select * from comdb2_partial_datacopies where tablename='t';

select * from t where a < 40;
select a, b, c from t where a < 40;

drop table t;
