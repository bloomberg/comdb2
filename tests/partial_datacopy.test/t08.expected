(tablename='t', keyname='A', keynumber=0, isunique='Y', isdatacopy='N', isrecnum='N', condition=NULL, uniqnulls='N', ispartialdatacopy='Y')
(tablename='t', keyname='A', columnname='b')
(tablename='t', keyname='A', columnname='c')
(rows inserted=25)
(a=1, b='cdb2', c=3, d=4)
(a=5, b='cdb6', c=7, d=8)
(a=9, b='cdb10', c=11, d=12)
(a=13, b='cdb14', c=15, d=16)
(a=17, b='cdb18', c=19, d=20)
(a=21, b='cdb22', c=23, d=24)
(a=25, b='cdb26', c=27, d=28)
(a=29, b='cdb30', c=31, d=32)
(a=33, b='cdb34', c=35, d=36)
(a=37, b='cdb38', c=39, d=40)
(a=1, b='cdb2', c=3)
(a=5, b='cdb6', c=7)
(a=9, b='cdb10', c=11)
(a=13, b='cdb14', c=15)
(a=17, b='cdb18', c=19)
(a=21, b='cdb22', c=23)
(a=25, b='cdb26', c=27)
(a=29, b='cdb30', c=31)
(a=33, b='cdb34', c=35)
(a=37, b='cdb38', c=39)
(id=3, parent=0, notused=0, detail='SEARCH TABLE t USING COVERING INDEX $A_28079FBF (a<?) (~262144 rows)')
(tablename='t', keyname='A', keynumber=0, isunique='Y', isdatacopy='N', isrecnum='N', condition=NULL, uniqnulls='N', ispartialdatacopy='Y')
(tablename='t', keyname='A', columnname='b')
(tablename='t', keyname='A', columnname='c')
(a=1, b='cdb2', c=3, d=4)
(a=5, b='cdb6', c=7, d=8)
(a=9, b='cdb10', c=11, d=12)
(a=13, b='cdb14', c=15, d=16)
(a=17, b='cdb18', c=19, d=20)
(a=21, b='cdb22', c=23, d=24)
(a=25, b='cdb26', c=27, d=28)
(a=29, b='cdb30', c=31, d=32)
(a=33, b='cdb34', c=35, d=36)
(a=37, b='cdb38', c=39, d=40)
(a=1, b='cdb2', c=3)
(a=5, b='cdb6', c=7)
(a=9, b='cdb10', c=11)
(a=13, b='cdb14', c=15)
(a=17, b='cdb18', c=19)
(a=21, b='cdb22', c=23)
(a=25, b='cdb26', c=27)
(a=29, b='cdb30', c=31)
(a=33, b='cdb34', c=35)
(a=37, b='cdb38', c=39)
(id=3, parent=0, notused=0, detail='SEARCH TABLE t USING COVERING INDEX $A_28079FBF (a<?) (~262144 rows)')
